aleph_export/do_extra_processing.py

```diff
@@ -0,0 +1,86 @@
+# do_extra_processing.py
+import os
+import json
+
+
+def do_extra_processing(value: str, extra_processing: str) -> str:
+    """ If extra processing is required, make appropriate calls to perform that additional processing. """
+    results = ""
+    if extra_processing == "link_to_source":
+        results = "https://onesearch.library.nd.edu/primo-explore/fulldisplay?docid=ndu_aleph" + value + "&context=L&vid=NDU&lang=en_US&search_scope=malc_blended&adaptor=Local%20Search%20Engine&tab=onesearch&query=any,contains,ndu_aleph002097132&mode=basic"  # noqa: E501
+    elif extra_processing == "lookup_work_type":
+        results = _lookup_work_type(value)
+    elif extra_processing == "format_subjects":
+        results = _format_subjects(value)
+    elif extra_processing == "format_creators":
+        results = _format_creators(value)
+    elif extra_processing == "translate_repository":
+        results = _translate_repository(value)
+    return results
```

Comment:
A slick way to handle this would be to use the local namespace
```python
    if extra_processing == "link_to_source":
        return "https://onesearch.library.nd.edu/primo-explore/fulldisplay?docid=ndu_aleph" + value + "&context=L&vid=NDU&lang=en_US&search_scope=malc_blended&adaptor=Local%20Search%20Engine&tab=onesearch&query=any,contains,ndu_aleph002097132&mode=basic"  # noqa: E501
    else:
        return locals()[("_" + extra_processing)](value)
```

Reply:
I hadn't seen this done.  I like it.  Thanks for the tip.

Reply:
Unfortunately, this didn't work.  I'm guessing if this had been a class, it may have worked.  Instead locals() only listed the parameters passed.  So I changed the code back to what I had originally had.